### DISCUSSION TITLE
Would it be possible to convert modified versions of melon ds to android

### DISCUSSION THREAD

**Discussion by Organizationguy** [rafaelvcaetano/melonDS-android#1276] [General]
How difficult would it be to conert a heavily altered version of Melon DS to play specific games?

**Comment by rafaelvcaetano** [comment]
It is possible. The effort depends on extent the modifications that were made and how they were made. Basically, you would need to replace the contents of the `melonds-android-lib` folder with your modifications. But you would need to keep some changes that I have also made there. From the top of my head, they are:
* In the root `CMakeLists.txt`
* The `src/android` folder
* Changes in the `Savestate.cpp` and `Savestate.h` files (the implementation was abstracted to support in-memory and file save-states)
* Some changes in the OpenGL implementation

But there are possibly many more minor differences. Plus, my implementation of melonDS is a bit behind the PC version. And I know that they have been making a lot of changes in the structure of the project. So, if your changes have been made on top of their recent changes, it would be a lot of work. If not, you can take a look at the actual differences that I have made [here](https://github.com/melonDS-emu/melonDS/compare/master...rafaelvcaetano:melonDS-android-lib:master) and see if you could adapt your changes while keeping mine.
